java/google/registry/rde/imports/RdeContactReader.java

```diff
@@ -80,23 +85,31 @@ public RdeContactReader(
   }
 
   @Override
-  public ContactResource next() throws IOException {
+  public JaxbFragment<XjcRdeContactElement> next() throws IOException {
     if (count < maxResults) {
-      if (parser == null) {
-        parser = newParser();
-        if (parser.isAtContact()) {
-          count++;
-          return XjcToContactResourceConverter.convertContact(parser.getContact());
+      try {
```

Comment:
Correct me if I am wrong, but I think the only thing  in this block that throws XmlException is readContact(). If that's so, move the try/catch into the helper method, here and in the other files with this pattern.